financial-connections/src/main/java/com/stripe/android/financialconnections/features/linkaccountpicker/LinkAccountPickerViewModel.kt

```diff
@@ -353,21 +354,41 @@ internal data class LinkAccountPickerState(
     data class Payload(
         val title: String,
         val accounts: List<Pair<PartnerAccount, NetworkedAccount>>,
-        val dataAccessNotice: DataAccessNotice?,
         val addNewAccount: AddNewAccount,
-        val merchantDataAccess: MerchantDataAccessModel,
         val consumerSessionClientSecret: String,
         val defaultCta: String,
         val nextPaneOnNewAccount: Pane?,
         val partnerToCoreAuths: Map<String, String>?,
         val singleAccount: Boolean,
+        val multipleAccountTypesSelectedDataAccessNotice: DataAccessNotice?,
+        val aboveCta: String?,
     ) {
 
         fun selectedPartnerAccounts(selected: List<String>): List<PartnerAccount> {
             return accounts.filter { it.first.id in selected }.map { it.first }
         }
     }
 
+    val activeDataAccessNotice: DataAccessNotice?
+        get() {
+            val payload = payload()
+            // Bank accounts can have multiple types
+            // (ex. linked account "bctmacct", manual account "csmrbankacct").
+            val selectedAccountTypes = selectedAccountIds
+                .mapNotNull { it.split("_").firstOrNull() }.toSet()
+            return if (selectedAccountTypes.size > 1) {
+                // if user selected multiple different account types, present a special data access notice
+                payload?.multipleAccountTypesSelectedDataAccessNotice
+            } else {
+                // we get here if user selected:
+                // 1) one account
+                // 2) or, multiple accounts of the same account type
+                payload?.accounts
+                    ?.firstOrNull { it.first.id in selectedAccountIds }
```

Comment:
Nit: You could add a `selectedAccounts` computed property to make this here a little more succinct. I also believe we can remove the comments, as the code explains what’s happening.